internal/mock/mocks.go

```diff
@@ -60,39 +65,40 @@ var (
 )
 
 type Repository struct {
-	ResolveResponse                notation.Descriptor
-	ResolveError                   error
-	ListSignatureManifestsResponse []registry.SignatureManifest
-	ListSignatureManifestsError    error
-	GetResponse                    []byte
-	GetError                       error
+	ResolveResponse            ocispec.Descriptor
+	ResolveError               error
+	ListSignaturesResponse     []ocispec.Descriptor
+	ListSignaturesError        error
+	FetchSignatureBlobResponse []byte
+	FetchSignatureBlobError    error
 }
 
 func NewRepository() Repository {
 	return Repository{
-		ResolveResponse: ImageDescriptor,
-		ListSignatureManifestsResponse: []registry.SignatureManifest{{
-			Blob:        JwsSigEnvDescriptor,
-			Annotations: Annotations,
-		}},
-		GetResponse: MockCaValidSigEnv,
+		ResolveResponse:            ImageDescriptor,
+		ListSignaturesResponse:     []ocispec.Descriptor{SigManfiestDescriptor},
+		FetchSignatureBlobResponse: MockCaValidSigEnv,
 	}
 }
 
-func (t Repository) Resolve(ctx context.Context, reference string) (notation.Descriptor, error) {
+func (t Repository) Resolve(ctx context.Context, reference string) (ocispec.Descriptor, error) {
 	return t.ResolveResponse, t.ResolveError
 }
 
-func (t Repository) ListSignatureManifests(ctx context.Context, manifestDigest digest.Digest) ([]registry.SignatureManifest, error) {
-	return t.ListSignatureManifestsResponse, t.ListSignatureManifestsError
+func (t Repository) ListSignatures(ctx context.Context, desc ocispec.Descriptor, fn func(signatureManifests []ocispec.Descriptor) error) error {
+	err := fn(t.ListSignaturesResponse)
+	if err != nil {
+		return err
+	}
```

Comment:
Yes. If `err == nil`, but we still want to mock ListSignaturesError, we will ` return t.ListSignaturesError` instead of nil. This is for unit test purpose.